tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/migration/AutoCorrectAction.java

```diff
@@ -112,6 +112,17 @@ public IStatus runInWorkspace( IProgressMonitor monitor )
                         }
                     }
 
+                    if( file.getFileExtension().equals( "jsp" ) || file.getFileExtension().equals( "jspf" ) )
+                    {
+                        try
+                        {
+                            Thread.sleep( 2000 );
```

Comment:
Why do we need to sleep?  This means we have a bad design some way, we shouldn't have to sleep to make it work.  How do we know 2 seconds is enough?  Maybe there are so many jsps and it takes 10 seconds?  can you explain what the problem is actually and we can redesign to avoid the need to sleep.